src/internal.c

```diff
@@ -30096,6 +30097,33 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
             }
         }
 
+#if defined(WOLFSSL_DTLS) && !defined(WOLFSSL_DTLS_HVR_ON_RESUME)
+        if (IsDtlsNotSctpMode(ssl) && IsDtlsNotSrtpMode(ssl) && !IsSCR(ssl)) {
+
+            if (!ssl->options.resuming) {
+
+                if (peerCookieSz != cookieSz ||
+                    XMEMCMP(peerCookie, newCookie, cookieSz) != 0) {
+                    /* Send newCookie to client in a HelloVerifyRequest message
+                     * and let the state machine alone. */
+                    ssl->msgsReceived.got_client_hello = 0;
+                    ssl->keys.dtls_handshake_number = 0;
+                    ssl->keys.dtls_expected_peer_handshake_number = 0;
+                    ssl->options.clientState = 0;
+
+                    *inOutIdx = begin + helloSz;
+
+                    ret = InitHandshakeHashes(ssl);
+                    if (ret != 0)
+                        goto out;
+
+                    ret = SendHelloVerifyRequest(ssl, newCookie, cookieSz);
+                    goto out;
+                }
```

Comment:
Consider refactoring this into a separate function since it appears twice in this function.

Reply:
I moved the common part inside the `SendHelloVerifyRequest()`